cmd/otel-allocator/allocation/strategy/state.go

```diff
@@ -0,0 +1,91 @@
+package strategy
```

Comment:
I think we should move the code in the `strategy` package back to `allocation`. `CollectorJSON` is only used in `allocation/http.go` and the other JSON structs are also around specifically for the http responses. It would also make naming make more sense (`allocation.New()` instead of `strategy.New()`).  It feels like we're splitting up code that we don't really need to split up.  I think that separating the code in this way makes it a little more difficult to navigate and figure out what's going on.

Reply:
I agree, i think a flat strategy here makes more sense for this codebase as it allows us to only export what's necessary